core/iwasm/common/wasm_c_api.c

```diff
@@ -281,24 +288,23 @@ static void
 wasm_engine_delete_internal(wasm_engine_t *engine)
 {
     if (engine) {
-        if (engine->stores) {
-            bh_vector_destroy((Vector *)engine->stores);
-            wasm_runtime_free(engine->stores);
-        }
-
         /* clean all created wasm_module_t and their locks */
-        {
-            unsigned i;
-            for (i = 0; i < engine->modules.num_elems; i++) {
-                wasm_module_ex_t *module;
-                if (bh_vector_get(&engine->modules, i, &module)) {
-                    os_mutex_destroy(&module->lock);
-                    wasm_runtime_free(module);
-                }
+        unsigned i;
+
+        for (i = 0; i < engine->modules.num_elems; i++) {
+            wasm_module_ex_t *module;
+            if (bh_vector_get(&engine->modules, i, &module)) {
+                os_mutex_destroy(&module->lock);
+                wasm_runtime_free(module);
             }
-            bh_vector_destroy(&engine->modules);
         }
 
+        bh_vector_destroy(&engine->modules);
+
+#ifndef os_thread_local_attribute
+        bh_vector_destroy(&engine->stores_by_tid);
+#endif
+
```

Comment:
The original code will destroy and free the engine->stores:
```C
        if (engine->stores) {
            bh_vector_destroy((Vector *)engine->stores);
            wasm_runtime_free(engine->stores);
        }
```
Wondering why the code is removed? If it isn't used, we should remove all the code related to engine->stores.

Reply:
`engine->stores` is not necessary any more. So I intend to remove all